ui/src/app/kvm/views/KVMDetails/KVMConfiguration/KVMConfiguration.test.tsx

```diff
@@ -0,0 +1,138 @@
+import { act } from "react-dom/test-utils";
+import { MemoryRouter, Route } from "react-router-dom";
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import React from "react";
+
+import {
+  pod as podFactory,
+  podState as podStateFactory,
+} from "testing/factories";
+
+import KVMConfiguration from "./KVMConfiguration";
+
+const mockStore = configureStore();
+
+describe("KVMConfiguration", () => {
+  let initialState;
+
+  beforeEach(() => {
+    const pods = [podFactory({ id: 1, name: "pod1" })];
+    const podState = podStateFactory({ items: pods, loaded: true });
+    initialState = {
+      config: {
+        items: [],
+      },
+      pod: podState,
+      resourcepool: {
+        items: [],
+        loaded: true,
+      },
+      tag: {
+        items: [],
+        loaded: true,
+      },
+      zone: {
+        items: [],
+        loaded: true,
+      },
+    };
+  });
+
+  it("fetches the necessary data on load", () => {
+    const state = { ...initialState };
+    const store = mockStore(state);
+    mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
+        >
+          <KVMConfiguration />
+        </MemoryRouter>
+      </Provider>
+    );
+    const expectedActions = [
+      "FETCH_POD",
+      "FETCH_RESOURCEPOOL",
+      "FETCH_TAG",
+      "FETCH_ZONE",
+    ];
+    const actions = store.getActions();
+    expectedActions.forEach((expectedAction) => {
+      expect(actions.some((action) => action.type === expectedAction));
+    });
+  });
+
+  it("displays a spinner if data has not loaded", () => {
+    const state = { ...initialState };
+    state.pod.loaded = false;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
+        >
+          <KVMConfiguration />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("Spinner").length).toBe(1);
+  });
+
+  it("can handle updating a KVM", () => {
```

Comment:
I think it'd be good to add a test for a non-kvm as well as it send slightly different data.

Reply:
Good idea, one for lxd and one for virsh